operator/internal/central/extensions/reconcile_tls_test.go

```diff
@@ -581,11 +587,186 @@ func Test_checkCertRenewal(t *testing.T) {
+				require.NotContains(t, new, mtls.SecondaryCACertFileName, "secondary CA cert should be removed")
+				require.NotContains(t, new, mtls.SecondaryCAKeyFileName, "secondary CA key should be removed")
+			},
+		},
+		{
+			name:   "no rotation action, secondary CA not present",
+			action: carotation.NoAction,
+			assert: func(t *testing.T, old, new types.SecretDataMap) {
+				require.Equal(t, old[mtls.CACertFileName], new[mtls.CACertFileName], "primary CA cert should be unchanged")
+				require.Equal(t, old[mtls.CAKeyFileName], new[mtls.CAKeyFileName], "primary CA key should be unchanged")
+				require.NotContains(t, new, mtls.SecondaryCACertFileName, "secondary CA cert should be present")
+				require.NotContains(t, new, mtls.SecondaryCAKeyFileName, "secondary CA key should be present")
+				require.NotEqual(t, old[mtls.ServiceCertFileName], new[mtls.ServiceCertFileName], "central cert should be reissued")
+				require.NotEqual(t, old[mtls.ServiceKeyFileName], new[mtls.ServiceKeyFileName], "central key should be reissued")
+			},
+		},
+		{
+			name:   "no rotation action, secondary CA present",
+			action: carotation.NoAction,
+			additionalSetup: func(t *testing.T, old types.SecretDataMap) {
+				secondary, err := certgen.GenerateCA()
+				require.NoError(t, err)
+				certgen.AddSecondaryCAToFileMap(old, secondary)
+			},
+			assert: func(t *testing.T, old, new types.SecretDataMap) {
+				require.Equal(t, old[mtls.CACertFileName], new[mtls.CACertFileName], "primary CA cert should be unchanged")
+				require.Equal(t, old[mtls.CAKeyFileName], new[mtls.CAKeyFileName], "primary CA key should be unchanged")
+				require.Equal(t, old[mtls.SecondaryCACertFileName], new[mtls.SecondaryCACertFileName], "secondary CA cert should be unchanged")
+				require.Equal(t, old[mtls.SecondaryCAKeyFileName], new[mtls.SecondaryCAKeyFileName], "secondary CA key should be unchanged")
+				require.NotEqual(t, old[mtls.ServiceCertFileName], new[mtls.ServiceCertFileName], "central cert should be reissued")
+				require.NotEqual(t, old[mtls.ServiceKeyFileName], new[mtls.ServiceKeyFileName], "central key should be reissued")
+			},
+		},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			primary, err := certgen.GenerateCA()
+			require.NoError(t, err)
+			oldFileMap := make(types.SecretDataMap)
+			certgen.AddCAToFileMap(oldFileMap, primary)
+			err = certgen.IssueCentralCert(oldFileMap, primary, mtls.WithNamespace("stackrox"))
+			require.NoError(t, err)
+
+			if tt.additionalSetup != nil {
+				tt.additionalSetup(t, oldFileMap)
+			}
+
+			r := &createCentralTLSExtensionRun{
+				centralObj:       &platform.Central{ObjectMeta: metav1.ObjectMeta{Namespace: "stackrox"}},
+				currentTime:      time.Now(),
+				caRotationAction: tt.action,
+				ca:               primary,
+			}
+
+			newFileMap, err := r.generateCentralTLSData(oldFileMap)
```

Comment:
This test will be super confused if `generateCentralTLSData` ever starts mutating the old file map.
Should we at least protect against it by cloning the map and optionally also guard against it by comparing it did not change during the call?

Reply:
Good point, I added this check.